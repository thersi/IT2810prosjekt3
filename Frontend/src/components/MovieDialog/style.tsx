import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(() => ({
  root: {
    background: "rgb(46, 140, 240)",
    padding: 10,
    maxWidth: "80%"
  },

  title: {
    fontSize: 22,
    color: "white",
    padding: 10,
    topMargin: 0,
    height: 0,
    textAlign: "center",
  },


  img: {
    flex: 1,
    maxWidth: "100%",
    maxHeight: "100%",
    height: "auto",
    margin: "auto",
    padding: "15%",
  },

  genres: {
    fontSize: 14,
    color: "white",
    padding: 10,
    textAlign: "center",
  },

  year: {
    fontSize: 14,
    color: "white",
    padding: 10,
    textAlign: "center",
  },

  button2: {
    color: "white",
    background: "rgb(46, 140, 240)",
    position: "absolute",
    border: "0",
  },

  thumb: {
    color: "white",
    fontSize: "medium",
    padding: 10,
    background: "rgb(46, 140, 240)",
    border: "0",
    
  },


  clickedThumb: {
    color: "gray",
    fontSize: "large",
    padding: 10,

  },
}));

export default useStyles;
