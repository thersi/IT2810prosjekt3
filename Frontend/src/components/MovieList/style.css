.movieContainer {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  gap: 10px;
  align-content: center;
  justify-content: center;
  padding-left: 5%;
  padding-right: 5%;
}

