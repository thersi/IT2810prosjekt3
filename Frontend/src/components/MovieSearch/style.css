.pagination {
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    margin: 15px;
    justify-content: center;
    align-content: center;
}