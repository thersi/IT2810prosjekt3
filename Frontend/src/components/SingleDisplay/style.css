
.movie {
  display: flex;
  padding: 3px;
  border: 2px outset;
  border-color: rgba(46, 140, 240, 0.1);
  background-color: rgba(46, 140, 240, 0.1);
  color: rgb(19, 48, 78);
  width: 70%;
  font-family: Arial, Helvetica, sans-serif;
}
@media screen and (max-width: 600px) {
  .movie {
    width: 95%;
  }
}

.movieInfo {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-left: 3px;
}

img {
  text-align: center;
  flex-basis: 40%;
  display: block;
  width: 120px;
  height: 160px;
}

.text {
  padding-left: 2%;
  font-size: 20px;
  flex-direction: column;
  display: block;
  width: 100%;
  height: 100%;
  padding-top: 3%;
}

.thumbButtons {
  display: flex;
  flex-direction: flex-end;

}
.thumbUp {
  padding: 20pt;
  color: rgb(125, 143, 125);
}
.thumbDown {
  padding: 20pt;
  color: rgb(175, 131, 131);
}
