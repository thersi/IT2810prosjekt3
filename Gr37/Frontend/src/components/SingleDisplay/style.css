.movie {
  display: flex;
  align-items: left;
  justify-content: left;
}

img {
  text-align: center;
  flex-basis: 40%;
  display: block;
  max-width: 90%;
}

.text {
  font-size: 20px;
  flex-direction: column;
  display: block;
}
