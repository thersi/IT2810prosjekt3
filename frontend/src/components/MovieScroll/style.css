.movieRow {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  gap: 20px;
  row-gap: 40px;
  overflow: hidden;
}
.movieContainer {
}
