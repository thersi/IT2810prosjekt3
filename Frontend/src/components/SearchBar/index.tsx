import SearchIcon from "@mui/icons-material/Search";
import { Search, SearchIconWrapper, StyledInputBase } from "./style";
import { searchProps } from "../../Interfaces/Interfaces";
import { ChangeEvent } from "react";

export default function SearchBar(props: searchProps) {
  /* Function that collects information tha user writes into search bar */
  const handleWrite = (
    e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => {
    props.handleSearch(e.target.value);
  };
  return (
    <Search>
      <SearchIconWrapper>
        <SearchIcon />
      </SearchIconWrapper>
      <StyledInputBase
        id="search-bar"
        placeholder="Movie Titles..."
        inputProps={{ "aria-label": "search" }}
        onChange={(e) => handleWrite(e)}
      />
    </Search>
  );
}
