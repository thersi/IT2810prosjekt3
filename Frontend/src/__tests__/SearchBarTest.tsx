import EnzymeToJson from "enzyme-to-json";
import { configure, mount, render, shallow } from "enzyme";
import { StyledInputBase } from "../components/SearchBar/style";
import Adapter from "enzyme-adapter-react-16";
import SearchBar from "../components/SearchBar";
import { fireEvent } from "@testing-library/dom/types/events";

const component = (
  <SearchBar
    handleSearch={() => {
      return "";
    }}
  />
);
configure({ adapter: new Adapter() });
describe("Search Bar", () => {
  it("renders correctly", () => {
    render(component);
  });
  it("Should match snapshot", () => {
    const snapshotCheck = shallow(component);
    expect(EnzymeToJson(snapshotCheck)).toMatchSnapshot();
  });

  it("Should contain components", () => {
    const shallowComp = shallow(component);
    const textInput = shallowComp.find("#search-bar");
    expect(textInput).toBeEmptyDOMElement;
    /* fireEvent.change(textInput, "Test"); */
    /*  shallowComp.simulate("click"); */
  });
});
