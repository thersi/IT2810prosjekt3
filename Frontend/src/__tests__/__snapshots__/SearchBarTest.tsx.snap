// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar Should match snapshot 1`] = `
<Styled(div)>
  <Styled(div)>
    <Memo(ForwardRef(SearchIcon)) />
  </Styled(div)>
  <Styled(ForwardRef(InputBase))
    id="search-bar"
    inputProps={
      Object {
        "aria-label": "search",
      }
    }
    onChange={[Function]}
    placeholder="Movie Titles..."
  />
</Styled(div)>
`;
